txscript/data/script_tests.json

```diff
@@ -1026,7 +1026,81 @@
 ["42", "SIZE 1 EQUALVERIFY 42 EQUAL", "NONE", "OK", "SIZE does not consume argument"],
 ["NOP", "SIZE 1", "NONE", "ERR_INVALID_STACK_OPERATION"],
 
-["Right bit rotation related test coverage"],
+["Bitwise logic opcode tests"],
+
+["INVERT test coverage"],
+["1", "INVERT -2 EQUAL", "NONE", "OK"],
+["0", "INVERT -1 EQUAL", "NONE", "OK"],
+["", "INVERT 1", "NONE", "ERR_INVALID_STACK_OPERATION", "INVERT requires an input parameter"],
+["'z'{2049}", "INVERT 1", "NONE", "ERR_PUSH_SIZE", "Exceed max parameter size"],
+
+["AND test coverage"],
+["1 0", "AND 0 EQUAL", "NONE", "OK"],
+["1 1", "AND 1 EQUAL", "NONE", "OK"],
+["'z'{2049} 0", "AND 1 EQUAL", "NONE", "ERR_PUSH_SIZE", "Exceed max parameter size"],
+["1", "AND 1", "NONE", "ERR_INVALID_STACK_OPERATION", "AND requires two operands"],
+
+["OR test coverage"],
+["1 0", "OR 1 EQUAL", "NONE", "OK"],
+["1 1", "OR 1 EQUAL", "NONE", "OK"],
+["'z'{2049} 0", "OR 1", "NONE", "ERR_PUSH_SIZE", "Exceed max parameter size"],
+["1", "OR 1", "NONE", "ERR_INVALID_STACK_OPERATION", "OR requires two operands"],
+
+["XOR test coverage"],
+["1 0", "XOR 1 EQUAL", "NONE", "OK"],
```

Comment:
Same here regarding many more tests for the numeric semantics being needed here as well as comments.